sdk/streamlib-python-wheel/python/streamlib/testing.py

```
@@ -136,6 +138,24 @@ def _build_and_start(self) -> None:
+                timeout=GRAPH_READY_TIMEOUT_SECONDS
+            )
+        except BaseException:
+            # A graph that never came up usually never started: the run loop
+            # raised on another thread and left every processor where it was.
+            # That failure is the cause; this one is the symptom.
+            try:
+                run_failure = self._run_failure.get_nowait()
+            except queue.Empty:
+                run_failure = None
+            if run_failure is not None:
+                raise run_failure
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)